### ISSUE TITLE
Type of "validator" is partially unknown

### ISSUE THREAD

**Issue by AlexanderPodorov** [pydantic/pydantic#5747]
### Initial Checks

- [X] I have searched GitHub for a duplicate issue and I'm sure this is something new
- [X] I have searched Google & StackOverflow for a solution and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this is a bug
- [X] I am confident that the issue is with pydantic (not my code, or another library in the ecosystem like [FastAPI](https://fastapi.tiangolo.com) or [mypy](https://mypy.readthedocs.io/en/stable))

### Description

Pylance in "strict" mode throws a warning when `pydantic.validator` is imported and used (the consequence of the first issue). I believe Pylance complains about incomplete `classmethod` signature.

### Example Code

```Python
from pydantic import validator

# throws:
# Type of "validator" is partially unknown
# Type of "validator" is "(*fields: str, pre: bool = False, each_item: bool = False, always: bool # = False, check_fields: bool = True, whole: bool | None = None, allow_reuse: bool = False) ->   
# ((((...) -> Any)) -> classmethod[Any, (...), Unknown])
```

### Python, Pydantic & OS Version

```Text
pydantic version: 1.10.7
pydantic compiled: True
python version: 3.11.3 (v3.11.3:f3909b8bc8, Apr  4 2023, 20:12:10) [Clang 13.0.0 (clang-1300.0.29.30)]
platform: macOS-13.0.1-arm64-arm-64bit
optional deps. installed: ['dotenv', 'email-validator', 'typing-extensions']
```

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [ ] [Data validation/parsing](https://docs.pydantic.dev/usage/models/#basic-model-usage)
- [ ] [Data serialization](https://docs.pydantic.dev/usage/exporting_models/) - `.dict()` and `.json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/usage/schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/usage/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/usage/model_config/)
- [ ] [Field Types](https://docs.pydantic.dev/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/usage/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/usage/models/) - `construct()`, pickling, private attributes, ORM mode
- [X] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by dmontagu** [comment]
I am not confident, but I think this is currently fixed on the `1.10.X-fixes` branch (which will be used to produce the 1.10.8 release), and is caused by the change to the type of `classmethod` in typeshed.

It would help if you could confirm that installing the `1.10.X-fixes` branch locally makes the issue goes away; if not we can still try to address it before the next (and possibly final) patch release.

**Comment by AlexanderPodorov** [comment]
Yes, it should work, I replaced the old signature:
```python
AnyClassMethod = classmethod[Any]
```
with the updated one:
```python
AnyClassMethod = classmethod[Any, Any, Any]
```
Thank you, closing the issue.
